pkg/auth/authz_handler.go

```diff
@@ -51,34 +53,28 @@ func (a *AuthzHandler) CreateAuthorizer() func(*http.Request) error {
 // and the principal was stored in the context in the "AuthKey" context value.
 func (a *AuthzHandler) Authorizer(request *http.Request) error {
 	payload := PayloadFromContext(request.Context())
-	username := payload.Username
-	payloadFromCache, found := a.client.Cache.Get(username)
```

Comment:
Why did you remove the cache?
The cache should only be on the allowedToUseAssistedInstaller() call